app/models/manageiq/providers/azure/cloud_manager/provision/cloning.rb

```diff
@@ -186,12 +186,7 @@ def start_clone(clone_options)
       vms = ::Azure::Armrest::VirtualMachineService.new(azure)
       vm  = vms.create(dest_name, resource_group.name, clone_options)
 
-      {
-        :subscription_id   => azure.subscription_id,
-        :vm_resource_group => vm.resource_group,
-        :type              => vm.type.downcase,
-        :vm_name           => vm.name
-      }
+      File.join(azure.subscription_id, vm.resource_group.downcase, vm.type.downcase, vm.name)
```

Comment:
NOTE this same logic lives in 3 different places now, need to refactor to reduce the duplication

Reply:
Feels like it could be a method on the vm itself taking the subscription_id as a param.

Reply:
Well `vm` here is the azure object not a `VmOrTemplate` instance since this is right after creating and before the record is created

Reply:
If all you need is a string, why not just `vm.id`?

Reply:
This method is supposed to return the new vm's ems_ref so this has to match https://github.com/ManageIQ/manageiq-providers-azure/blob/master/app/models/manageiq/providers/azure/inventory/parser/cloud_manager.rb#L67-L70 exactly